include/pros/optical.hpp

```diff
@@ -0,0 +1,232 @@
+/**
+ * \file pros/optical.hpp
+ *
+ * Contains prototypes for functions related to the VEX Optical sensor.
+ *
+ * Visit https://pros.cs.purdue.edu/v5/tutorials/topical/imu.html to learn
+ * more.
+ *
+ * This file should not be modified by users, since it gets replaced whenever
+ * a kernel upgrade occurs.
+ *
+ * Copyright (c) 2017-2020, Purdue University ACM SIGBots.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef _PROS_OPTICAL_HPP_
+#define _PROS_OPTICAL_HPP_
+
+#include <stdbool.h>
+#include <cstdint>
+#include "pros/optical.h"
+
+namespace pros {
+class Optical {
+	public:
+  /**
+	 * Creates an Optical Sensor object for the given port.
+	 *
+	 * This function uses the following values of errno when an error state is
+	 * reached:
+	 * ENXIO - The given value is not within the range of V5 ports (1-21).
+	 * ENODEV - The port cannot be configured as an Optical Sensor
+	 *
+	 * \param port
+	 *        The V5 port number from 1-21
+	 */
+  explicit Optical(const std::uint8_t port);
+
+
+	/**
+	 * Get the detected color hue
+   *
+	 * This is not avaliable if gestures are being detected. Hue has a
+	 * range of 0 to 359.999
+	 *
+	 * This function uses the following values of errno when an error state is
+	 * reached:
+	 * ENXIO - The given value is not within the range of V5 ports (1-21).
+	 * ENODEV - The port cannot be configured as an Optical Sensor
+	 *
+	 * \return hue value if the operation was successful or PROS_ERR_F if the operation
+	 * failed, setting errno.
+	 */
+	double get_hue();
```

Comment:
This is a question for @HotelCalifornia. We usually make these virtual, but I'm not entirely sure why.

Reply:
Should this be virtual for this PR?

Reply:
Private and non-virtual are the defensive choices. We can do that and see if anyone asks for the need to extend them.

Reply:
private and all virtual is what we have done, so just do that for now.